internal/provider/organization_resource.go

```diff
@@ -0,0 +1,198 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ resource.Resource = (*organizationResource)(nil)
+
+func NewOrganizationResource() resource.Resource {
+	return &organizationResource{}
+}
+
+type organizationResource struct {
+	client *ExtendedGqlClient
+}
+
+type organizationResourceModel struct {
+	OrgId       types.String `tfsdk:"id"`
+	Name        types.String `tfsdk:"name"`
+	OrgMrn      types.String `tfsdk:"mrn"`
+	Description types.String `tfsdk:"description"`
+}
+
+func (r *organizationResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_organization"
+}
+
+func (r *organizationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				MarkdownDescription: "Name of the space.",
+				Required:            true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(
+						regexp.MustCompile(`^([a-zA-Z \-'_]|\d){2,30}$`),
+						"must contain 2 to 30 characters, where each character can be a letter (uppercase or lowercase), a space, a dash, an underscore, or a digit",
+					),
+				},
+			},
+			"id": schema.StringAttribute{
+				MarkdownDescription: "ID of the space. Must be globally unique. If the provider has a space configured and this field is empty, the provider space is used.",
+				Optional:            true,
```

Comment:
Can this id be specified by the user? or shy is this optional?

Reply:
we should try to use it if its specified and available